### PR TITLE
dhclient set hostname tweaks

### PR DESCRIPTION
**Pull request by mchf**
[bsc#1013605](https://bugzilla.suse.com/show_bug.cgi?id=1013605) openQA test fails: Hostname Error Pop-Up During Installation

\+ listbox is enabled only when wicked service is selected

### CODE REVIEW

File: src/include/network/services/dns.rb [hunk 1]
```diff
@@ -430,7 +431,7 @@ def InitDhcpHostname(_key)
         fix_dhclient_warning(LanItems.invalid_dhcp_cfgs)
 
         selected = NO_CHANGE_LABEL
-        items << [Item(Id(NO_CHANGE_LABEL), _("keep current settings"), true)]
+        items << Item(Id(NO_CHANGE_LABEL), _("keep current settings"), true)
```
Comment (mvidner) [inline src/include/network/services/dns.rb]: `- fixed displaying the "keep current settings" option`

File: src/lib/network/network_autoyast.rb [hunk 2]
```diff
@@ -22,6 +22,19 @@ def initialize
       Yast.import "Linuxrc"
     end
 
+    # Clears ifcfg files to be ready for proposals done later by the installer
+    #
+    # @return [void]
+    def clear_ifcfgs
```
Comment (mvidner) [inline src/lib/network/network_autoyast.rb]: The name and the comment make it sound like the entire contents of the ifcfgs are emptied. Compare http://www.rubydoc.info/search/stdlib/core?q=clear

File: src/include/network/services/dns.rb (hunk 1 above)
Comment (mchf) [inline src/include/network/services/dns.rb]: Ah this was the one pointed during meeting. I didn't check that yet :-/ Good to know that it is going to be fixed too ;-)

File: src/lib/network/network_autoyast.rb (hunk 2 above)
Comment (mchf) [inline src/lib/network/network_autoyast.rb]: ok

**Comment by mchf** [comment]
1) I plan to touch linuxrc too - learning opportunity 
2) I don't know If I found all flows which can lead to such inconsistent confirg - one never know what can linuxrc do next time, so
3) I think that implementing that in yast makes installer more robust. Especially when all new code fits into approx. five lines

**Comment by mvidner** [comment]
Proposed linuxrc fix: https://github.com/openSUSE/linuxrc/pull/131

**Comment by okurz** [comment]
hi, the corresponding linuxrc PR has been merged but we still see problems like e.g. https://openqa.opensuse.org/tests/324003#step/yast2_lan/12
I assume this PR here intends to fix it?

**Comment by nilxam** [comment]
ah, linuxrc update is pending in staging A now https://build.opensuse.org/request/show/446728 , will get it into Factory asap

**Comment by mchf** [comment]
@okurz no, the fix was moved into linuxrc completely. This PR fixes some minor issues only.

**Comment by mvidner** [comment]
Closing this one in favor of #477.

@okurz no, that problem is because the linuxrc fix is not yet being used.

**Pull request closed without merging**